### PR TITLE
Conversion tests

### PR DESCRIPTION
**Pull request by dinges2**

### CODE REVIEW

**Review by nielsm5 — APPROVED** [review]

File: core/src/test/java/nl/nn/adapterframework/xslt/Xslt3Test.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
+package nl.nn.adapterframework.xslt;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
```
Comment (tnleeuw) [inline core/src/test/java/nl/nn/adapterframework/xslt/Xslt3Test.java:5]: Shall we move these tests to JUnit5 before merging or do you want to just get it merged?

**Review by tnleeuw — APPROVED** [review]

File: core/src/test/java/nl/nn/adapterframework/xslt/Xslt3Test.java (hunk 1 above)
Comment (nielsm5) [reply core/src/test/java/nl/nn/adapterframework/xslt/Xslt3Test.java:5]: I contemplated that, I thought we might need to backport this to the 7.8 and thus it's better to leave it at JUnit 4 for now..

File: core/src/main/java/nl/nn/adapterframework/pipes/ForEachChildElementPipe.java [hunk 2]
```diff
@@ -511,7 +511,7 @@ public void setElementXPathExpression(String string) {
 	}
 
 	/**
-	 * If set to <code>2</code> a XSLT processor 2.0 (net.sf.saxon) will be used, supporting XPath 2.0, otherwise a XSLT processor 1.0 (org.apache.xalan), supporting XPath 1.0. N.B. Be aware that setting this other than 1 might cause the input file being read as a whole in to memory, as XSLT Streaming is currently only supported by the XSLT Processor that is used for xsltVersion=1
+	 * If set to <code>2</code> a XSLT processor 2.0 (net.sf.saxon) will be used, supporting XPath 2.0, otherwise a XSLT processor 1.0 (org.apache.xalan), supporting XPath 1.0. N.B. Be aware that setting this other than 1 might cause the input file being read as a whole in to memory, as XSLT Streaming is currently only supported by the XSLT Processor that is used for xsltVersion=1. This pipe supports up to and including XSLT version 3.0
```
Comment (tnleeuw) [inline core/src/main/java/nl/nn/adapterframework/pipes/ForEachChildElementPipe.java:514]: Actually this comment should be updated to say that the Saxon processor will be used when setting this parameter to 2 or 3.

File: core/src/main/java/nl/nn/adapterframework/parameters/Parameter.java [hunk 3]
```diff
@@ -945,7 +945,7 @@ public void setXpathExpression(String xpathExpression) {
 	}
 
 	/**
-	 * when set to <code>2</code> xslt processor 2.0 (net.sf.saxon) will be used, otherwise xslt processor 1.0 (org.apache.xalan). <code>0</code> will auto detect
+	 * when set to <code>2</code> xslt processor 2.0 (net.sf.saxon) will be used, otherwise xslt processor 1.0 (org.apache.xalan). <code>0</code> will auto detect. This class supports up to and including XSLT version 3.0.
```
Comment (tnleeuw) [inline core/src/main/java/nl/nn/adapterframework/parameters/Parameter.java:948]: Actually this comment should be updated to say that the Saxon processor will be used when setting this parameter to 2 or 3.

File: core/src/main/java/nl/nn/adapterframework/pipes/XmlSwitch.java [hunk 4]
```diff
@@ -244,7 +244,7 @@ public void setEmptyForwardName(String empty){
 	}
 
 	/**
-	 * If set to <code>2</code> xslt processor 2.0 (net.sf.saxon) will be used, otherwise xslt processor 1.0 (org.apache.xalan). <code>0</code> will auto detect
+	 * If set to <code>2</code> xslt processor 2.0 (net.sf.saxon) will be used, otherwise xslt processor 1.0 (org.apache.xalan). <code>0</code> will auto detect. This pipe supports up to and including XSLT version 3.0
```
Comment (tnleeuw) [inline core/src/main/java/nl/nn/adapterframework/pipes/XmlSwitch.java:247]: Actually this comment should be updated to say that the Saxon processor will be used when setting this parameter to 2 or 3.

File: core/src/main/java/nl/nn/adapterframework/senders/XsltSender.java [hunk 5]
```diff
@@ -456,7 +456,7 @@ public void setSkipEmptyTags(boolean b) {
 	}
 
 	/**
-	 * If set to <code>1</code> xslt processor 1.0 (org.apache.xalan) will be used, otherwise xslt processor 2.0 (net.sf.saxon). <code>0</code> will auto detect
+	 * If set to <code>1</code> xslt processor 1.0 (org.apache.xalan) will be used, otherwise xslt processor 2.0 (net.sf.saxon). <code>0</code> will auto detect. This class supports up to and including XSLT version 3.0
```
Comment (tnleeuw) [inline core/src/main/java/nl/nn/adapterframework/senders/XsltSender.java:459]: Actually this comment should be updated to say that the Saxon processor will be used when setting this parameter to 2 or 3.

File: core/src/main/java/nl/nn/adapterframework/pipes/XmlIf.java [hunk 6]
```diff
@@ -184,7 +184,7 @@ public void setRegex(String regex){
 	}
 
 	/**
-	 * specifies the version of xslt to use
+	 * specifies the version of xslt to use. This pipe supports up to and including XSLT version 3.0
```
Comment (tnleeuw) [inline core/src/main/java/nl/nn/adapterframework/pipes/XmlIf.java:187]: Perhaps the doc for version here should be same as in other classes that have an XsltVersion argument?

**Review by tnleeuw — CHANGES_REQUESTED** [review]
I think actually some of the JavaDoc comments should be updated to better reflect how an XSLT implementation version is chosen.

**Review by tnleeuw — APPROVED** [review]

**Pull request merged**